src/comestible_inv_pane.h

```diff
@@ -0,0 +1,177 @@
+         */
+        void paginate();
+
+        void add_items_from_area( comestible_inv_area *area, bool from_cargo, units::volume &ret_volume,
+                                  units::mass &ret_weight );
+        virtual comestible_inv_listitem *create_listitem( std::list<item *> list, int index,
+                comestible_inv_area *area, bool from_vehicle ) = 0;
+        virtual comestible_inv_listitem *create_listitem( const item_category *cat = nullptr ) = 0;
+
+        void print_items();
+
+        /** Scroll to next non-header entry */
+        void skip_category_headers( int offset );
+        /** Only add offset to index, but wrap around! */
+        void mod_index( int offset );
+
+        void clear_items();
+
+        mutable std::map<std::string, std::function<bool( const item & )>> filtercache;
+
+    public:
+
+        virtual ~comestible_inventory_pane();
+
+        /**
+         * Index of the selected item (index of @ref items),
+         */
+        int index;
+        catacurses::window window;
+        //this filter is supplied by parent, applied before user input filtering
+        std::function<bool( const item & )> special_filter;
+        //list of shortcut reminders (created manually by parent)
+        std::string title;
+        std::vector<legend_data> additional_info;
+
+        int itemsPerPage;
+
+        bool inCategoryMode;
+
+        comestible_inv_columns sortby;
+        // secondary sort, in case original returns == items
+        comestible_inv_columns default_sortby;
+
+        /**
+         * Whether to recalculate the content of this pane.
+         * Implies @ref redraw.
+         */
+        bool needs_recalc;
+        bool needs_redraw;
+
+        virtual void init( int items_per_page, catacurses::window w,
+                           std::array<comestible_inv_area, comestible_inv_area_info::NUM_AIM_LOCATIONS> *a );
+        void save_settings( bool reset );
+
+        /**
+         * Makes sure the @ref index is valid (if possible).
+         */
+        void fix_index();
+        void recalc();
+        void redraw();
+
+        /**
+         * Scroll @ref index, by given offset, set redraw to true,
+         * @param offset Must not be 0.
+         */
+        void scroll_by( int offset );
+        /**
+         * @return either null, if @ref index is invalid, or the selected
+         * item in @ref items.
+         */
+        comestible_inv_listitem *get_cur_item_ptr();
+        /**
+         * Set the filter string
+         */
+        void set_filter( const std::string &new_filter );
+        void start_user_filtering( int h, int w );
+
+        //adds things this can sort on. List is provided by parent
+        void add_sort_entries( uilist &sm );
+
+        std::array<comestible_inv_area, comestible_inv_area_info::NUM_AIM_LOCATIONS> *all_areas;
+        comestible_inv_area *get_square( comestible_inv_area_info::aim_location loc ) {
+            return  &( ( *all_areas )[loc] );
```

Comment:
Why return a pointer? It can never be null. This should return a reference. Same for the below.